package/MDAnalysis/core/Selection.py

```diff
@@ -178,6 +178,12 @@ def __init__(self, parser, tokens):
         pass
 
     def apply(self, group):
+        # Check whether group is identical to the one stored
+        # in the corresponding universe, in which case this
+        # is returned directly. This works since the Universe.atoms
+        # are unique by construction.
+        if group.universe and group is group.universe.atoms:
```

Comment:
This probably isn't doing what you think it is.

```python
if True and False is False:
    print 'This gets printed'
```
Because the implicit brackets are `(if True) and (False is False)`.

Doesn't just `if group is group.universe.atoms` work?